hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/serialization/PriorityQueueHook.java

```diff
@@ -59,7 +59,7 @@ public void write(ObjectDataOutput out, PriorityQueue queue) throws IOException
             @Override
             public PriorityQueue read(ObjectDataInput in) throws IOException {
                 int size = in.readInt();
-                PriorityQueue res = new PriorityQueue(size, in.readObject());
+                PriorityQueue res = new PriorityQueue(Math.min(1, size), in.readObject());
```

Comment:
Isn't that suppose to be `Math.max(1, size)` ?

Reply:
It was fixed here: https://github.com/hazelcast/hazelcast-jet/pull/1172